data/transactions/logic/eval.go

```diff
@@ -2000,6 +2022,34 @@ func checkBranch(cx *EvalContext) error {
 	cx.branchTargets[target] = true
 	return nil
 }
+
+// checks any branch that is {op} {int16 be offset}
+func checkSwitch(cx *EvalContext) error {
+	// first call to get the number of offsets, 0 is a safe choice because there must exist at least one label
+	_, numOffsets, err := branchSwitch(cx, 0)
+	if err != nil {
+		return err
+	}
+
+	_, bytesUsed := binary.Uvarint(cx.program[cx.pc+1:])
```

Comment:
Since `branchSwitch(cx, 0)` worked, we don't need to check `bytesUsed` again here, but a comment to effect would be good.